django/views/decorators/cache.py

```diff
@@ -26,41 +31,73 @@ def cache_page(timeout, *, cache=None, key_prefix=None):
     )
 
 
+@sync_and_async_middleware
 def cache_control(**kwargs):
     def _cache_controller(viewfunc):
-        @wraps(viewfunc)
-        def _cache_controlled(request, *args, **kw):
+        def _process_request(request):
             # Ensure argument looks like a request.
             if not hasattr(request, "META"):
                 raise TypeError(
                     "cache_control didn't receive an HttpRequest. If you are "
                     "decorating a classmethod, be sure to use "
                     "@method_decorator."
                 )
-            response = viewfunc(request, *args, **kw)
+
+        def _process_response(response, **kwargs):
             patch_cache_control(response, **kwargs)
```

Comment:
This is a single line. I don't see the benefit of the `_process_response()` wrapper.